src/main/java/org/influxdb/impl/InfluxDBImpl.java

```diff
@@ -99,34 +124,94 @@ public InfluxDBImpl(final String url, final String username, final String passwo
     setLogLevel(LOG_LEVEL);
 
     this.gzipRequestInterceptor = new GzipRequestInterceptor();
+    client.addInterceptor(loggingInterceptor).addInterceptor(gzipRequestInterceptor);
+
+    Factory converterFactory = null;
+    ObjectMapper mapper = null;
+    if (useMsgPack) {
+      client.addInterceptor(chain -> {
+        Request request = chain.request().newBuilder().addHeader("Accept", APPLICATION_MSGPACK).build();
+        return chain.proceed(request);
+      });
+
+      MessagePackFactory messagePackFactory = new MessagePackFactory();
+      ExtensionTypeCustomDeserializers extTypeCustomDeserializers = new ExtensionTypeCustomDeserializers();
+      final byte msgPackTimeExtType = (byte) 5;
+      final int timeOffset = 0;
+      final int timeByteArrayLength = 8;
+      extTypeCustomDeserializers.addCustomDeser(msgPackTimeExtType, data -> {
+        return ByteBuffer.wrap(data, timeOffset, timeByteArrayLength).getLong();
+      });
+      messagePackFactory.setExtTypeCustomDesers(extTypeCustomDeserializers);
+
+      mapper = new ObjectMapper(messagePackFactory);
+      //mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+      converterFactory = MessagePackConverterFactory.create(mapper);
+    } else {
+      mapper = new ObjectMapper(new MappingJsonFactory());
+      converterFactory = MoshiConverterFactory.create();
+    }
+
+    JavaType javaType = mapper.getTypeFactory().constructType(QueryResult.class);
+    final ObjectReader objectReader =  mapper.readerFor(javaType);
+
+    chunkProccesor = (chunkedBody, consumer) -> {
+        BufferedSource source = chunkedBody.source();
+
+        QueryResult result = null;
+        MappingIterator<QueryResult> results = objectReader.readValues(source.inputStream());
+        while (results.hasNext()) {
+          result = results.nextValue();
+          consumer.accept(result);
+        }
+    };
```

Comment:
Way too much code in the constructor.

Reply:
In a minor release, I would like to keep the implementation as back-ward compatible as possible. this constructor

```
  InfluxDBImpl(final String url, final String username, final String password, final OkHttpClient.Builder client,
          final InfluxDBService influxDBService, final JsonAdapter<QueryResult> adapter) {
```
in my opinion, even it seems awkward and used nowhere in the source code, we are not used if it is used by any user or not so I decided to keep it. That lead me to introduce the ChunkProcessor so we can have polymorphism for processing __chunking query__

@majst01, I believe Jackson has a better API on processing HTTP chunked response